### PR TITLE
Only tries to parse the month variables if the data exists

### PR DESCRIPTION
**Pull request by b-m-f**
# Done
Updates an if statement to be more thorough

### CODE REVIEW

File: canonicalwebteam/blog/logic.py [hunk 1]
```diff
@@ -97,13 +97,15 @@ def transform_article(
             article["content"]["rendered"]
         )
 
-    if "_start_month" in article:
+    # if _start_month is set here we can assume that day and year
+    # will also have a value, same for the end data
+    if "_start_month" in article and article["_start_month"] != "":
```
Comment (jpmartinspt) [inline canonicalwebteam/blog/logic.py]: Can this not just be article.get("start_month")?

Comment (b-m-f) [reply canonicalwebteam/blog/logic.py]: Thanks! updated it to use `""` as a default value on get, to catch both the None and empty string scenario

File: canonicalwebteam/blog/logic.py [hunk 2]
```diff
@@ -97,13 +97,15 @@ def transform_article(
             article["content"]["rendered"]
         )
 
-    if "_start_month" in article:
+    # if _start_month is set here we can assume that day and year
+    # will also have a value, same for the end data
+    if article.get("_start_month", "") != "":
```
Comment (tbille) [inline canonicalwebteam/blog/logic.py]: no need for default value here.

File: canonicalwebteam/blog/logic.py [hunk 3]
```diff
@@ -97,13 +97,15 @@ def transform_article(
             article["content"]["rendered"]
         )
 
-    if "_start_month" in article:
+    # if _start_month is set here we can assume that day and year
+    # will also have a value, same for the end data
```
Comment (tbille) [inline canonicalwebteam/blog/logic.py]: no need to comment a if statement.

And assuming is not a good pattern

File: canonicalwebteam/blog/logic.py (hunk 2 above)
Comment (tbille) [reply canonicalwebteam/blog/logic.py]: ` if article.get("_start_month"):`

File: canonicalwebteam/blog/logic.py [hunk 4]
```diff
@@ -97,13 +97,15 @@ def transform_article(
...
+    if article.get("_start_month", "") != "":
         start_month_name = get_month_name(int(article["_start_month"]))
         article["start_date"] = "{} {} {}".format(
             article["_start_day"], start_month_name, article["_start_year"]
         )
 
-    if "_end_month" in article:
+    if article.get("_end_month", "") != "":
```
Comment (tbille) [inline canonicalwebteam/blog/logic.py]: same here

File: canonicalwebteam/blog/logic.py (hunk 3 above)
Comment (b-m-f) [reply canonicalwebteam/blog/logic.py]: Do you prefer a test on all attributes here?

This is not documented in the [wordpress Rest docs](https://developer.wordpress.org/rest-api/reference/posts/) as far as I can tell. Seems to be done with a plugin.

Comment (tbille) [reply canonicalwebteam/blog/logic.py]: then what do you trust more?

Comment (b-m-f) [reply canonicalwebteam/blog/logic.py]: Alright, I think this speaks for a full check then? Updated it.

**Pull request merged**